### ISSUE TITLE
System workqueue misuse and misleading documentation

### ISSUE THREAD

**Issue by GrantRolls** [zephyrproject-rtos/zephyr#61819]
In the documentation for the workqueues, the overall outline and suggested uses are misleading.
The suggest use is defined as
` "defer complex interrupt-related processing from an ISR to a shared thread. This allows the interrupt-related processing to be done promptly without compromising the system’s ability to respond to subsequent interrupts, and does not require the application to define and manage an additional thread to do the processing."`, which is great.

The issues comes about because the documentation heavily suggests to NOT create another work queue due to resource footprint `"Additional workqueues should only be defined when it is not possible to submit new work items to the system workqueue"`, which at the outsets puts you into the view you should ONLY try to use system workqueue - it is not clear why or when you shouldn't use the system workqueue.

Any complex operation inevitably requires the use of some form of semaphore handling to deal with shared data.
The actuality is that so many underlying systems are using this workqueue, that having ANY blocking in a worker task means that those underlying systems get stalled and can cause numerous issues. 

An example is using a worker on the system workqueue to schedule bluetooth notify sends - because the underlying notify operation relies on the workqueue to send, release buffers and indicate success, it quickly falls over if the calling worker blocks to wait for success.

It is only through trial and error that you realise that application level code should never rely on the system workqueue.

**Describe the solution you'd like**
I would like to see several changes

Documentation
* Updates to the documentation to raise the high-use and importance of the system workqueue, in underlying engine code
* Change to the documentation to indicate that alternate work queues should be used for any application level code

Features
* A secondary out-of-the-box work queue, like APPLICATION_WORK_QUEUE, so that all projects are not needing to roll their own

**Comment by porthengrath-di** [comment]
I agree. System Workqueue is used too heavily in Zephyr subsystem such as BLE stack, Sensors' driver, LTE stack, and SMP. 

I think there should be at least a watchdog implementation to detect if system work queue is blocked for too long.

**Comment by brianwyld** [comment]
It should at least be possible to provide a different workqueue to underlying drivers (example, the mcp23xxx gpio driver) so that their time critical processing (handling interrupt from mcp to do I2C operations to read the IO in a timely fashion) are not blocked by other system code (BLE stack, etc etc). In particular it is not documented exactly which drivers/modules/subsystems use the system workqueue, so its trial and error to understand why the gpio functions get blocked in certain cases!
